controller/src/main/java/io/pravega/controller/store/PravegaTablesScope.java

```diff
@@ -180,37 +213,73 @@ public String getName() {
+                                                                                        .collect(Collectors.toList());
+                                    return CompletableFuture.allOf(storeHelper.deleteTable(streamsInScopeTableName, true, context.getRequestId()),
+                                                                   storeHelper.deleteTable(kvtsInScopeTableName, true, context.getRequestId()),
+                                                                   storeHelper.deleteTable(rgsInScopeTableName, true, context.getRequestId()),
+                                                                   Futures.allOf(deleteTagTablesFut))
+                                                            .thenAccept(v -> log.debug("tables deleted {} {} {}", streamsInScopeTableName,
+                                                                                       kvtsInScopeTableName, rgsInScopeTableName));
+                                })
+                                .thenCompose(deleted -> storeHelper.removeEntry(SCOPES_TABLE, scopeName, context.getRequestId()));
     }
 
     @Override
-    public CompletableFuture<Pair<List<String>, String>> listStreams(int limit, String continuationToken, Executor executor, 
+    public CompletableFuture<Pair<List<String>, String>> listStreams(int limit, String continuationToken, Executor executor,
                                                                      OperationContext context) {
         Preconditions.checkNotNull(context, "Operation context cannot be null");
         return getStreamsInScopeTableName(context)
                 .thenCompose(streamsInScopeTable -> readAll(limit, continuationToken, streamsInScopeTable, context));
     }
 
+    @Override
+    public CompletableFuture<Pair<List<String>, String>> listStreamsForTag(String tag, String continuationToken, Executor executor,
+                                                                           OperationContext context) {
+
+        Preconditions.checkNotNull(context, "Operation context cannot be null");
+
+        return getNextTableSegment(tag, continuationToken, context).thenCompose(pair -> {
+            if (pair.getLeft().isEmpty() && !pair.getRight().contains(LAST_TAG_CHUNK)) {
+                return listStreamsForTag(tag, pair.getRight(), executor, context);
+            } else {
+                return CompletableFuture.completedFuture(pair);
+            }
+        });
+    }
+
+    CompletableFuture<Pair<List<String>, String>> getNextTableSegment(String tag, String ct, OperationContext context) {
+        return getStreamTagsInScopeTableName(context).thenApply(
+                strings -> {
+                    if (ct.isEmpty()) {
+                        return strings.get(0);
+                    } else {
+                        // return next index
+                        return strings.get(strings.indexOf(ct) + 1);
```

Comment:
i dont quite follow the logic here? what does strings contain -- all the chunk ids?
and we are saying we want to read from the next chunk? right?
perhaps adding some comments in the code explaining the logic here would be good.

Reply:
Ok, I have improved the code and added comments here.